__tests__/V5ClusterManagement.js

```diff
@@ -0,0 +1,458 @@
+  expect(getByText(newNodePoolName)).toBeInTheDocument();
+
+  // Is it now the 2nd node pool in list?
+  const reSortedNodePools = getAllByTestId('node-pool-id');
+  expect(reSortedNodePools[1]).toHaveTextContent(nodePoolId);
+
+  // Assert that the mocked responses got called, tell them to stop waiting for
+  // a request.
+  nodePoolPatchRequest.done();
+});
+
+// TODO This test triggers a memory leak error related with setting state depending
+// on the response of an asynchronous call in ScaleNodePoolModal.
+// Not fixing it now because is a "minor" error, this error can't break the app and
+// because I will be working on the data flow refactor that will solve this.
+it(`shows the v5 cluster scaling modal when the button is clicked with default values and 
+scales node pools correctly`, async () => {
+  // TODO default values from constants file
+  const defaultScaling = { min: 3, max: 10 };
+  const increaseValue = 1;
+  const newScaling = {
+    min: defaultScaling.min + increaseValue,
+    max: defaultScaling.max + increaseValue,
+  };
+
+  const nodePool = nodePoolsResponse[0];
+  const nodePoolPatchResponse = {
+    ...nodePoolsResponse[0],
+    scaling: newScaling,
+  };
+
+  // Request
+  const nodePoolPatchRequest = nock(API_ENDPOINT)
+    .intercept(
+      `/v5/clusters/${V5_CLUSTER.id}/nodepools/${nodePool.id}/`,
+      'PATCH'
+    )
+    .reply(200, nodePoolPatchResponse);
+
+  const div = document.createElement('div');
+  const {
+    getByText,
+    getAllByText,
+    getAllByTestId,
+    getByLabelText,
+  } = renderRouteWithStore(ROUTE, div, {});
+
+  await wait(() => getAllByTestId('node-pool-id'));
+
+  // Expect first nodePool is the first one in the list.
+  const nodePoolId = getAllByTestId('node-pool-id')[0].textContent;
+  expect(nodePoolId).toBe(nodePool.id);
+
+  fireEvent.click(getAllByText('•••')[0]);
+  fireEvent.click(getByText(/edit scaling limits/i));
+
+  await wait(() => getByText(/edit scaling settings for/i));
+
+  // Is the modal in the document?
+  const modalTitle = getByText(/edit scaling settings for/i);
+  expect(modalTitle).toBeInTheDocument();
+  expect(modalTitle).toHaveTextContent(nodePool.id);
+
+  const inputMin = getByLabelText(/minimum/i);
+  const inputMax = getByLabelText(/maximum/i);
+
+  // Are the correct values in the correct fields?
+  expect(inputMin.value).toBe(defaultScaling.min.toString());
+  expect(inputMax.value).toBe(defaultScaling.max.toString());
+
+  // Change the values and modify the scaling settingsw
+  fireEvent.change(inputMin, { target: { value: newScaling.min } });
+  fireEvent.change(inputMax, { target: { value: newScaling.max } });
+  const textButton = `Increase minimum number of nodes by ${newScaling.min}`;
+
+  await wait(() => getByText(textButton));
+
+  const submitButton = getByText(textButton);
+  fireEvent.click(submitButton);
+
+  //Wait for the Flash message to appear
+  await wait(() => {
+    getByText(/succesfully edited node pool name/i);
```

Comment:
We're scaling here, but expecting a message that confirms a name change?

Reply:
Indicates maybe something wrong in the code, not in the test.

Reply:
These flash messages are outside of the rendered `div` so they do not dissapear when component is unmounted after each test. What is happening here is that we are finding the flash message that triggered the test before.

So the only thing that it is wrong is that we are not expecting the right message.

I have tried to remove the flash message in the end of the tests when they are interfering with other tests.